packages/svelte-ux/src/lib/components/settings.ts

```diff
@@ -111,6 +125,37 @@ export function getSettings(): Settings {
   } catch (error) {
     return {
       currentTheme: createThemeStore({ light: ['light'], dark: ['dark'] }),
+      componentSettingsCache: {},
     };
   }
 }
+
+export function resolveComponentSettings<NAME extends ComponentName>(
+  settings: Settings,
+  name: NAME
+): ResolvedComponentSettings<NAME> {
+  const { classes: themeClasses, ...defaultProps } = settings?.components?.[name] ?? {};
+  const classes = resolveComponentClasses(themeClasses);
+
+  const output: ResolvedComponentSettings<NAME> = {
+    defaults: (defaultProps ?? {}) as ResolvedDefaultProps<NAME>,
+    classes,
+  };
+
+  return output;
+}
+
+export function getComponentSettings<NAME extends ComponentName>(
+  name: NAME
+): ResolvedComponentSettings<NAME> {
+  const settings = getSettings();
+
+  const existing = settings.componentSettingsCache[name];
```

Comment:
Using a cache will mean we can't change these at runtime, but not a big use case other than docs / demos.  The performance improvements (likely small, but add up being in the hot path) are worth it I think.

Reply:
Yeah you can't change them at runtime anyway since it's not a store. At least not without reaching directly into the object and then recreating any controls that you want to change, and in that case you might as well also just set the cache back to `{}` :)

When you call `settings` again it does create a new cache for that new instance of the settings object, so you can still do "sub-settings" like you do in a couple places.

Reply:
Awe, good point.  Hadn't really though tit through.  Thanks!